### PR TITLE
Enable dark mode #751 and move NavMenu logic to LayoutService #830

### PR DESCRIPTION
**Pull request by hocinehacherouf**
Enable dark mode #751 and move NavMenu logic to LayoutService #830

## Description

What's new?

- Implement #751 
- Implement #830 
- Fix some unit tests #827 

## What kind of change does this PR introduce?

- [x] Feature
- [x] Bugfix

### CODE REVIEW

File: src/AzureIoTHub.Portal.Server.Tests.Unit/Pages/DeviceConfigurations/DeviceConfigurationDetailPageTests.cs [hunk 1]
```diff
@@ -681,18 +682,20 @@ public void WhenClickToSaveShouldSendPutToTheEndpoint()
                     return true;
                 });
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Server.Tests.Unit/Pages/DeviceConfigurations/DeviceConfigurationDetailPageTests.cs:684]: Not part of this PR ?

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

File: src/AzureIoTHub.Portal.Server.Tests.Unit/Pages/DeviceConfigurations/DeviceConfigurationDetailPageTests.cs (hunk 1 above)
Comment (hocinehacherouf) [reply src/AzureIoTHub.Portal.Server.Tests.Unit/Pages/DeviceConfigurations/DeviceConfigurationDetailPageTests.cs:684]: I fixed a unit test that failed randomly on ci. Related to the issue #827

**Review by kbeaugrand — APPROVED** [review]
